platforms/ucp/setup/dynamic-slots/ucp-dynamic-slots.setup.ts

```diff
@@ -1,18 +1,20 @@
 import { DynamicSlotsSetup } from '@platforms/shared';
+import { context, insertNewSlot } from '@wikia/ad-engine';
 import { Injectable } from '@wikia/dependency-injection';
 
 @Injectable()
 export class UcpDynamicSlotsSetup implements DynamicSlotsSetup {
 	configureDynamicSlots(): void {
-		this.injectTopLeaderboardPlaceholder();
+		this.injectSlots();
 	}
 
-	private injectTopLeaderboardPlaceholder(): void {
-		const container = document.querySelector('.WikiaPage');
-		const topLeadeboardWrapper = document.createElement('div');
-		const parentElement = container.parentNode;
-
-		topLeadeboardWrapper.id = 'top_leaderboard';
-		parentElement.insertBefore(topLeadeboardWrapper, container);
+	private injectSlots(): void {
+		const slots = context.get('slots');
+		Object.values(slots).forEach((slot) => {
+			const { nextSibling, adProduct }: any = slot;
```

Comment:
Don't use `any` please. I'd rename this property to `nextSiblingSelector`. I'm not sure  what`adProduct` actually is. Use `slot.getSlotName()` please:
```suggestion
			const { nextSiblingSelector }: string = slot;
			
			if (nextSiblingSelector) {
				insertNewSlot(slot.getSlotName(), document.querySelector(nextSiblingSelector), true);
			}
```

Reply:
I've changed the function so it doesn't use adProduct now. Also I've fixed the types for slots

Reply:
I think you did not push your changes since I can't see them. I mean for example: using `slot.getSlotName()`.

Reply:
@Loczi94 it's my bad. `getSlotName` is not available here. It's not an `AdSlot` object, it's slot definition only.